nifi-nar-bundles/nifi-snmp-bundle/nifi-snmp-processors/src/main/java/org/apache/nifi/snmp/processors/GetSNMP.java

```diff
@@ -87,113 +91,110 @@
+    public void onTrigger(ProcessContext context, ProcessSession processSession) {
+        final String targetUri = snmpContext.getTarget().getAddress().toString();
+        final String snmpStrategy = context.getProperty(SNMP_STRATEGY).getValue();
+        final String oid = context.getProperty(OID).getValue();
+
+        if ("GET".equals(snmpStrategy)) {
+            final ResponseEvent response = snmpGetter.get();
+            if (response.getResponse() != null) {
                 PDU pdu = response.getResponse();
-                flowFile = SNMPUtils.updateFlowFileAttributesWithPduProperties(pdu, flowFile, processSession);
-                flowFile = SNMPUtils.addAttribute(SNMPUtils.SNMP_PROP_PREFIX + "textualOid",
-                        context.getProperty(TEXTUAL_OID).getValue(), flowFile, processSession);
-                processSession.getProvenanceReporter().receive(flowFile,
-                        this.snmpTarget.getAddress().toString() + "/" + context.getProperty(OID).getValue());
-                if(pdu.getErrorStatus() == PDU.noError) {
+                FlowFile flowFile = createFlowFile(context, processSession, pdu);
+                processSession.getProvenanceReporter().receive(flowFile, targetUri + "/" + oid);
+                if (pdu.getErrorStatus() == PDU.noError) {
                     processSession.transfer(flowFile, REL_SUCCESS);
                 } else {
                     processSession.transfer(flowFile, REL_FAILURE);
                 }
             } else {
-                this.getLogger().error("Get request timed out or parameters are incorrect.");
+                getLogger().error("Get request timed out or parameters are incorrect.");
                 context.yield();
             }
-        } else if("WALK".equals(context.getProperty(SNMP_STRATEGY).getValue())) {
-            final List<TreeEvent> events = this.targetResource.walk();
-            if((events != null) && !events.isEmpty() && (events.get(0).getVariableBindings() != null)) {
+        } else if ("WALK".equals(snmpStrategy)) {
+            final List<TreeEvent> events = snmpGetter.walk();
+            if (areValidEvents(events)) {
                 FlowFile flowFile = processSession.create();
                 for (TreeEvent treeEvent : events) {
                     flowFile = SNMPUtils.updateFlowFileAttributesWithTreeEventProperties(treeEvent, flowFile, processSession);
                 }
-                processSession.getProvenanceReporter().receive(flowFile,
-                        this.snmpTarget.getAddress().toString() + "/" + context.getProperty(OID).getValue());
+                processSession.getProvenanceReporter().receive(flowFile, targetUri + "/" + oid);
                 processSession.transfer(flowFile, REL_SUCCESS);
             } else {
-                this.getLogger().error("Get request timed out or parameters are incorrect.");
+                getLogger().error("Get request timed out or parameters are incorrect.");
```

Comment:
Both this error log and the error log for `GET` contain the same message.  It would be helpful to differentiate these two log messages with some additional parameters, including at least the strategy used.
```suggestion
                getLogger().error("SNMP WALK Failed: Request timed out or parameters may be invalid");
```